### DISCUSSION TITLE
How to include default initrd AND a custom one?

### DISCUSSION THREAD

**Discussion by septatrix** [systemd/mkosi#2660] [Q&A]
Is it possible to use the default initrd *and* a custom one? The document seems to suggest that this is not possible. As soon as `Initrds=` is specified the default initrd is no longer build.

I am aware that one can use `Include=mkosi-initrd` in ones initrd configuration but that is not equivalent as stuff like kernel modules, image-id/-version, hostname etc. are no longer taken from the main config.

**Comment by daandemeyer** [comment]
@septatrix Use `InitrdInclude=` to dump whatever extras you need into the default initrd.

**Reply by septatrix** [reply]
What we want in the end are two initrds/UKIs so this would not work. One of them would just be the normal one and another would function as an installer.

What would also work is to generate two UKIs from the default initrd but one of them with `rd.systemd.unit=installer.target` or `init=/installer.sh` appended to the `KernelCommandLine=` but I am not aware of any way to do this either.

**Reply by daandemeyer** [reply]
@septatrix Each UKI that mkosi generates gets the same initrds. The only thing that differs is the kernel modules. We do not support having different initrds for different UKIs
